src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java

```diff
@@ -106,6 +117,9 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeOptionalString(modelName);
         out.writeOptionalString(featureSetName);
         out.writeMap(params);
+        if (out.getVersion().onOrAfter(Version.V_6_2_4)) {
+            out.writeOptionalStringArray(activeFeatures != null ? activeFeatures.toArray(new String[0]) : null);
```

Comment:
nitpick: `activeFeatures.toArray(new String[activeFeatures.size()])`

Reply:
regarding initializing the array with zero sized element:  seems like, providing a zero-length array, even creating it and throwing it away, is on average faster than allocating an array of the right size as per this blogpost: https://shipilev.net/blog/2016/arrays-wisdom-ancients/

I am okay using your suggestion if this seems like an micro-optimization and you prefer to keep this codebase more consistent with initializing with `size()`

Reply:
interesting read thanks :)
I probably have bad habits so whenever I see something that could be initialized to the right size I point it. But please keep the code as you like and feel free to ignore this nitpick.